src/plugins/shikiCodeblocks/index.ts

```diff
@@ -0,0 +1,153 @@
+
+const themeNames = Object.keys(themes);
+const devIconCss = "@import url('https://cdn.jsdelivr.net/gh/devicons/devicon@v2.10.1/devicon.min.css');";
+
+const getSettings = () => Settings.plugins.ShikiCodeblocks as ShikiSettings;
+
+export default definePlugin({
+    name: "ShikiCodeblocks",
+    description: "Brings vscode-style codeblocks into Discord, powered by Shiki",
+    authors: [Devs.Vap],
+    patches: [
+        {
+            find: "codeBlock:{react:function",
+            replacement: {
+                match: /codeBlock:\{react:function\((.),(.),(.)\)\{/,
+                replace: "$&return Vencord.Plugins.plugins.ShikiCodeblocks.renderHighlighter($1,$2,$3);",
+            },
+        },
+    ],
+    start: async () => {
+        setStyle(cssText, StyleSheets.Main);
+        if (getSettings().useDevIcon !== DeviconSetting.Disabled)
+            setStyle(devIconCss, StyleSheets.DevIcons);
+
+        await shiki.init(getSettings().customTheme || getSettings().theme);
+    },
+    stop: () => {
+        shiki.destroy();
+        clearStyles();
+    },
+    options: {
+        theme: {
+            type: OptionType.SELECT,
+            description: "Default themes",
+            options: themeNames.map(themeName => ({
+                label: wordsToTitle(wordsFromPascal(themeName)),
+                value: themes[themeName],
+                default: themes[themeName] === themes.DarkPlus,
+            })),
+            disabled: () => !!getSettings().customTheme,
+            onChange: shiki.setTheme,
+        },
+        customTheme: {
+            type: OptionType.STRING,
+            description: "A link to a custom vscode theme",
+            placeholder: themes.MaterialCandy,
+            isValid: value => {
+                if (!value) return true;
+                const url = parseUrl(value);
+                if (!url) return "Must be a valid URL";
+
+                if (!url.pathname.endsWith(".json")) return "Must be a json file";
+
+                return true;
+            },
+            onChange: value => shiki.setTheme(value || getSettings().theme),
+        },
+        tryHljs: {
+            type: OptionType.SELECT,
+            description: "Use the more lightweight default Discord highlighter and theme.",
+            options: [
+                {
+                    label: "Never",
+                    value: HljsSetting.Never,
+                },
+                {
+                    label: "Secondary",
+                    value: HljsSetting.Secondary,
+                    default: true,
+                },
+                {
+                    label: "Primary",
+                    value: HljsSetting.Primary,
+                },
+                {
+                    label: "Always",
+                    value: HljsSetting.Always,
+                },
+            ],
+        },
+        useDevIcon: {
+            type: OptionType.SELECT,
+            description: "How to show devicons on codeblocks",
```

Comment:
```suggestion
            description: "How to show language icons on codeblocks",
```
(?) I'm not sure what devicons stands for

Reply:
its the library i use for lang icons, this is prob better since its user-facing text